lib/jsdom/living/helpers/style-rules.js

```diff
@@ -44,6 +45,58 @@ exports.forEachMatchingSheetRuleOfElement = (elementImpl, handleRule) => {
 
   handleSheet(parsedDefaultStyleSheet);
   forEach.call(elementImpl._ownerDocument.styleSheets._list, handleSheet);
+}
+
+const styleCache = new WeakMap();
```

Comment:
Instead of a weak map, why not put this on the document itself? Combined with the non-nullness of `_ownerDocument`, that could greatly reduce the lines of code needed to access the style cache. Just have the weak map always exist.